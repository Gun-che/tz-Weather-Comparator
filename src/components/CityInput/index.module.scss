.wrap {
  position: relative;
  padding-right: 40px;
  height: 60px;
  button {
    position: absolute;
    top: 5px;
    right: 0;
    line-height: 0;
  }
}
