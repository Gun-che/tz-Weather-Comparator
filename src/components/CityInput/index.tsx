import React, { SyntheticEvent } from 'react'
import s from './index.module.scss'

interface ICityInputProps {
  index: number;
  onChange(e: React.ChangeEvent<HTMLInputElement & { dataset: { index: number } }>): void;
  item: [string, number];
  onClose(index: number): (e: SyntheticEvent) => void;
}

export const CityInput: React.FC<ICityInputProps> = ({
  index,
  item,
  onChange,
  onClose
}) => {

  return (
    <div className={"input-field " + s.wrap}>
      <button
        className="btn-floating btn-small waves-effect waves-light blue lighten-3"
        onClick={onClose(index)}>
        <i role="button" className="material-icons">close</i>
      </button>
      <i className="material-icons prefix">location_on</i>
      <input
        className="validate"
        data-index={index}
        type="text"
        onChange={onChange}
        id={'city' + (index + 1)}
        name={'city' + (index + 1)}
        value={item[0]}
      />
      <label htmlFor={'city' + (index + 1)}>{index + 1 + ' город'}</label>
    </div>
  )
}
