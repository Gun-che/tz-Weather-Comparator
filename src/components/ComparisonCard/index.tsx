import React from 'react'
import { IWeatherData } from '../../types/state'
import s from './index.module.scss'
import img from './wp.jpg'

export const ComparisonField: React.FC<{ data: IWeatherData[] }> = ({ data }) => {

  const sortByTemp = (): [string, number, number] => {
    let temp = -50
    let feelsLike = 0
    let city = ''
    data.forEach(i => {
      if (i.main.feels_like > temp) {
        temp = i.main.temp;
        city = i.name;
        feelsLike = i.main.feels_like
      }
    })
    return [city, temp, feelsLike]
  }

  return (
    <div className={s.wrap}>
      <div className="card">
        <div className="card-image">
          <img src={img} alt="sky" />
          <span className="card-title">{sortByTemp()[0]} cамый тёплый город, из выбранных вами!</span>
        </div>
        <div className="card-content">
          <p>Температура здесь сегодня {Math.round(sortByTemp()[1]) + `°C!`}</p>
          <p>По ощущениям {Math.round(sortByTemp()[2]) + `°C`}</p>
        </div>

      </div>
      <h2>{}</h2>
    </div>
  )
}
