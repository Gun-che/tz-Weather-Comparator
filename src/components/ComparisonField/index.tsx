import React from 'react'
import { IWeatherData } from '../../types/state'

export const ComparisonField: React.FC<{ data: IWeatherData[] }> = ({ data }) => {
  return (
    <div>

      {data.length ? <h2>{data[0].main.temp} ? {data[1].main.temp}</h2> : null}
    </div>
  )
}
