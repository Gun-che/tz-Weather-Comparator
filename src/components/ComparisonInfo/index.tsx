import React from 'react'
import { IWeatherData } from '../../types/state'

interface ISortResult {
  warmCity: string;
  coldCity: string;
  windyCity: string;
  windlessCity: string;
  wetCity: string;
  dryCity: string;
}

export const ComparisonInfo: React.FC<{ data: IWeatherData[] }> = ({ data }) => {

  const sort = (): ISortResult => {
    let warm = -100
    let cold = 100
    let windy = 0
    let windless = 100
    let wet = 0
    let dry = 100
    let warmCity = ''
    let coldCity = ''
    let windyCity = ''
    let windlessCity = ''
    let wetCity = ''
    let dryCity = ''

    data.forEach(i => {

      if (i.main.feels_like > warm) {
        warmCity = i.name;
        warm = i.main.feels_like
      }

      if (i.main.feels_like < cold) {
        coldCity = i.name;
        cold = i.main.feels_like
      }

      if (i.wind.speed > windy) {
        windyCity = i.name;
        windy = i.main.feels_like
      }

      if (i.wind.speed < windless) {
        windlessCity = i.name;
        windless = i.main.feels_like
      }

      if (i.main.humidity > wet) {
        wetCity = i.name;
        wet = i.main.feels_like
      }

      if (i.main.humidity < dry) {
        dryCity = i.name;
        dry = i.main.feels_like
      }
    })

    return {
      warmCity,
      coldCity,
      windyCity,
      windlessCity,
      wetCity,
      dryCity,
    }
  }

  const {
    warmCity,
    coldCity,
    windyCity,
    windlessCity,
    wetCity,
    dryCity, }: ISortResult = sort()

  return (
    <div>
      <div className="card blue-grey darken-1">
        <div className="card-content white-text">
          <span className="card-title">Card Title</span>
          <p>Теплее прочего: {warmCity}</p>
          <p>Холоднее прочего: {coldCity}</p>
          <p>Ветренее прочего: {windyCity}</p>
          <p>Наиенее ветренно: {windlessCity}</p>
          <p>Самая высокая влажность: {wetCity}</p>
          <p>Суше прочего: {dryCity}</p>
        </div>
      </div>
    </div>
  )
}
