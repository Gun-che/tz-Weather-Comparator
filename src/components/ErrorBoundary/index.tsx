import React from 'react'
import s from './index.module.scss'

interface IState {
  hasError: boolean;
  msg: string;
  err: number;
}

export class ErrorBoundary extends React.Component<{}, IState> {
  constructor(props: {}) {
    super(props);
    this.state = {
      hasError: false,
      msg: '',
      err: 0,
    }
  }

  static getDerivedStateFromError(error: Error) {
    return {
      hasError: true,
      msg: error.message,
      err: error.name,
    }
  }

  render() {
    const err = this.state.err;
    const msg = this.state.msg;
    const hasError = this.state.err;

    if (hasError) {
      return (
        <>
          <h1>Что-то пошлло не так, пожалуйста обновите страницу!</h1>
          <div className={s.wrap + ' container'}>
            {err && err}
            {msg && <h3>{msg}</h3>}
          </div>
        </>
      )

    } else {
      return this.props.children
    }
  }
}
