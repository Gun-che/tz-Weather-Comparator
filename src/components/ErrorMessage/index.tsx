import React from 'react'
import s from './index.module.scss'

export const ErrorMessage: React.FC<{ message: string }> = (
  { message }
) => {

  return (<>
    {message ? (<div className={s.wrap + ' containerCust'}><h2>{message}</h2></div>) : (<div className={s.emptyWrap}></div>)
    }
  </>)
}
