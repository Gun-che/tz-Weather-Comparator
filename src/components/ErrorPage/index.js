import React from 'react'
import PropTypes from 'prop-types'
import * as s from './index.module.scss'

export default function ErrorPage({ err, msg }) {
  return (
    <>
      <div className={s.wrap + ' container'}>
        {err && err}
        {msg && <h3>{msg}</h3>}
      </div>
    </>
  )
}

ErrorPage.propTypes = {
  err: PropTypes.string.isRequired,
  msg: PropTypes.string.isRequired,
}

