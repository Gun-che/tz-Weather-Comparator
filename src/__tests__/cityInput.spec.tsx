import React from 'react'
import { shallow, mount } from 'enzyme'
import { CityInput, ICityInputProps } from '../components/CityInput'

describe('CityInput component', () => {
  const props: ICityInputProps = {
    index: 1,
    onChange: jest.fn(),
    item: ['q', 1],
    onClose: jest.fn(),
  }
  describe('CityInput component render', () => {
    const CityInputRender = shallow(<CityInput {...props} />)

    it('renders properly', () => {
      expect(CityInputRender).toMatchSnapshot()
    })

    it('render <button>', () => {
      expect(CityInputRender.find('button')).toHaveLength(1)
    })

    it('render <input>', () => {
      expect(CityInputRender.find('input')).toHaveLength(1)
    })

    it('render <label>', () => {
      expect(CityInputRender.find('label')).toHaveLength(1)
    })
  })
})