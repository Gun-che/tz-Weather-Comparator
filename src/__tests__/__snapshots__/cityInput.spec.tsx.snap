// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityInput component CityInput component render renders properly 1`] = `ShallowWrapper {}`;
